qcodes/tests/instrument_mocks.py

```diff
@@ -377,6 +388,37 @@ def get_raw(self):
         return item
 
 
+
+class ComplexArraySetPointParam(ArrayParameter):
+    """
+    Arrayparameter that returns complex numbers
+    """
+
+    def __init__(self, instrument=None, name='testparameter'):
+        shape = (5,)
+        label = 'this label'
+        unit = 'this unit'
+        sp_base = tuple(np.linspace(5, 9, 5))
+        setpoints = (sp_base,)
+        setpoint_names = ('this_setpoint',)
+        setpoint_labels = ('this setpoint',)
+        setpoint_units = ('this setpointunit',)
+        super().__init__(name,
+                         shape,
+                         instrument,
+                         label=label,
+                         unit=unit,
+                         setpoints=setpoints,
+                         setpoint_labels=setpoint_labels,
+                         setpoint_names=setpoint_names,
+                         setpoint_units=setpoint_units)
+
+    def get_raw(self):
+        item = np.arange(5) - 1j*np.arange(5)
+        self._save_val(item)
```

Comment:
unlike `ParameterWithSetpoints`, this save_val is needed for ArrayParameter? oh dear ...